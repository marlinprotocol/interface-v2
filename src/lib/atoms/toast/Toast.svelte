<script lang="ts">
	import { dismissToast, toastsStore } from '$lib/data-stores/toastStore';
	import { cn } from '$lib/utils/helpers/commonHelper';
	import { fade, slide } from 'svelte/transition';
</script>

{#if $toastsStore.length > 0}
	<div class="toast toast-end toast-top z-[9999] items-end p-10" data-testid="toast">
		{#each $toastsStore as toast (toast.id)}
			<button
				in:slide
				out:fade
				data-testid="toast-btn"
				on:click={() => dismissToast(toast.id)}
				class={cn(
					'alert flex w-fit flex-row items-start justify-start gap-3 whitespace-normal !rounded-[15px] border-0 py-[17.5px] pl-[13px] pr-[46px] font-medium shadow-lg',
					toast.className
				)}
			>
				<div class={'flex h-[54px] w-[54px] items-center justify-center rounded-full bg-white'}>
					<img src={toast.iconData} alt={toast.iconData} />
				</div>
				<span class="max-w-[330px] grow text-left text-sm">
					{#if toast.message.title}
						<div class="font-poppins text-lg text-[#000000]">{toast.message.title}</div>
					{/if}
					{#if toast.message.description}
						<p class="text-base font-light text-[#707070]">
							{toast.message.description}
						</p>
					{/if}
				</span>
			</button>
		{/each}
	</div>
{/if}
