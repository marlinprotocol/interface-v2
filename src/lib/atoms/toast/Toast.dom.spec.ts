import { cleanup, render } from '@testing-library/svelte';
import Toast from './Toast.svelte';
// import { chevronDown } from 'svelte-awesome/icons';
// import type { CommonVariant } from '$lib/types/componentTypes';

const { toastsStore } = await vi.hoisted(
	() => import('$lib/data-stores/mock-data-stores/mockStores')
);

// const data = {
// 	iconColor: 'red',
// 	message: 'You have successfully toast your notification',
// 	className: 'Default',
// 	dismissible: true,
// 	iconData: chevronDown,
// 	id: 298429842984,
// 	timeout: 2000,
// 	variant: 'primary' as CommonVariant,
// 	bgColor: 'red'
// };

beforeAll(() => {
	vi.mock('$lib/data-stores/toastStore', async () => {
		return {
			toastsStore,
			dismissToast: vi.fn()
		};
	});
});

describe('Toast', () => {
	afterEach(() => {
		cleanup();
		vi.resetAllMocks();
		toastsStore.dismissToast(298429842984);
	});

	it('renders no toasts initially', () => {
		const { queryAllByTestId } = render(Toast);
		expect(queryAllByTestId('toast')).toHaveLength(0);
	});

	it('renders a toast when added to the store', () => {
		// toastsStore.mockSetSubscribeValue([data]);
		// const { getByTestId } = render(Toast);
		// const toastElement = getByTestId('toast');
		// const toastClasslist = toastElement.classList;
		// expect(toastElement).toBeTruthy();
		// expect(toastClasslist.contains('toast')).toBe(true);
		// expect(toastElement).toMatchSnapshot();
	});

	it('fires the onClick event when clicked toast Icon', async () => {
		// toastsStore.mockSetSubscribeValue([data]);
		// const { getByTestId, queryAllByTestId } = render(Toast);
		// const toastBtn = getByTestId('toast-btn');
		// await fireEvent.click(toastBtn);
		// waitForElementToBeRemoved(queryAllByTestId('toast')).then(() =>
		// 	expect(queryAllByTestId('toast')).toHaveLength(0)
		// );
	});

	it('renders with toast class', () => {
		// toastsStore.mockSetSubscribeValue([data]);
		// const { getByTestId } = render(Toast);
		// expect(getByTestId('toast').classList.contains('toast')).toBe(true);
	});

	it('renders with toast position top', () => {
		// toastsStore.mockSetSubscribeValue([data]);
		// const { getByTestId } = render(Toast);
		// expect(getByTestId('toast').classList.contains('toast-top')).toBe(true);
	});
});
