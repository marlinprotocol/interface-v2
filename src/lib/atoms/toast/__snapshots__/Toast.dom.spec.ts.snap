// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Toast > renders a toast when added to the store 1`] = `
<div
  class="toast toast-end toast-top z-[9999] items-end"
  data-testid="toast"
>
  <button
    class="alert flex w-fit flex-row items-start justify-start gap-1 whitespace-normal rounded px-4 py-2 font-medium shadow-lg Default"
    data-testid="toast-btn"
  >
    <div
      class="flex h-5 w-5 items-center justify-center"
    >
      <svg
        aria-label=""
        class="fa-icon red s-T5YyQewRBd7n"
        data-testid="icon"
        height="16"
        role="presentation"
        style="width:18px; height:18px; cursor:pointer;  "
        version="1.1"
        viewBox="0 0 1792 1792"
        width="16"
      >
        <path
          d="M1683 808l-742 741q-19 19-45 19t-45-19l-742-741q-19-19-19-45.5t19-45.5l166-165q19-19 45-19t45 19l531 531 531-531q19-19 45-19t45 19l166 165q19 19 19 45.5t-19 45.5z"
        />
        
         
         
        
        
      </svg>
      <!--&lt;Svg&gt;-->
      <!--&lt;Icon&gt;-->
      <!--&lt;Icon&gt;-->
    </div>
     
    <span
      class="max-w-[330px] grow text-left text-sm"
    >
      You have successfully toast your notification
    </span>
     
  </button>
</div>
`;
